docs/pages/button.js

```diff
@@ -259,245 +229,481 @@ card(
+  />,
 );
 
 card(
-  <Example
-    name="Color: transparent and semiTransparentWhite"
-    id="transparent"
-    defaultCode={`function Example() {
-  const [selectedTransparent, setSelectedTransparent] = React.useState(false);
-  const [selectedSemiTransparentWhite, setSelectedSemiTransparentWhite] = React.useState(false);
-
-  return (
-    <Flex>
-      <Box column={4} padding={2}>
-        <Box margin={3}>
-          <Button
-            text="Toggle text color & compare contrast"
-            onClick={() => setSelectedTransparent((currentValue) => !currentValue)}
-            selected={selectedTransparent}/>
-        </Box>
-        <Image
-          alt="Image to compare contrast between text color and background."
-          color="rgb(231, 186, 176)"
-          naturalHeight={751}
-          naturalWidth={564}
-          src="https://i.ibb.co/7bQQYkX/stock2.jpg"
-        >
-          <Box padding={2}>
+  <MainSection name="Variants">
+    <MainSection.Subsection
+      title="Size"
+      description={`Button is available in 3 fixed sizes:
+1. \`lg\` (48px)
+    Large is the only size that should be used on Pinner surfaces.
+2. \`md\` (40px)
+    Medium is used on more dense UI such as business surfaces or internal tools.
+3. \`sm\` (32px)
+    Small should be used sparingly and only in places where the UI is very dense.`}
+    >
+      <CombinationNew size={['sm', 'md', 'lg']}>
+        {({ size }) => (
+          <Button accessibilityLabel={`Example size ${size}`} color="red" text="Save" size={size} />
+        )}
+      </CombinationNew>
+    </MainSection.Subsection>
+    <MainSection.Subsection
+      title="Width"
+      description={`
+1. Inline (default)
+    Inline is our default Button width.  The width of an inline Button is based on the length of its text. Use in most cases where you need a Button.
+2. Full-width (\`fullWidth\`)
+    Full-width Buttons can be used in narrower content areas when the text in the Button is close to full width in the content area. This is especially common to see in components such as Callout and Upsell at their smaller breakpoints.`}
+    >
+      <CombinationNew fullwidth={[false, true]}>
+        {({ fullwidth }) => (
           <Button
-            color={selectedTransparent ? 'transparentWhiteText' : 'transparent'}
-            iconEnd="add-pin"
-            text="Save this image"
+            accessibilityLabel={`Example width ${fullwidth}`}
+            color="red"
+            text="Save"
+            fullWidth={fullwidth}
+            size="lg"
           />
-          </Box>
-          <Box padding={2} position="absolute" bottom right>
+        )}
+      </CombinationNew>
+    </MainSection.Subsection>
+    <MainSection.Subsection
+      title="Color on white backgrounds"
+      description={`
+1. Red (Primary)
+    High emphasis, used for primary actions.
```

Comment:
can we give some hints of what's a primary/secondary/tertiary action?

Reply:
I think this is a great thing to add down the road because, to be honest, I think we need to do more thinking on this topic before we can provide a cogent answer..